src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -256,28 +253,27 @@ internal static string DeterminePrimaryScriptFile(JObject functionConfig, string
             return Path.GetFullPath(functionPrimary);
         }
 
-        private static ScriptType ParseScriptType(string scriptFilePath)
+        internal static string ParseLanguage(string scriptFilePath, IEnumerable<WorkerConfig> workerConfigs)
         {
+            // determine the script type based on the primary script file extension
             string extension = Path.GetExtension(scriptFilePath).ToLowerInvariant().TrimStart('.');
-
+            string language = string.Empty;
             switch (extension)
             {
                 case "csx":
                 case "cs":
-                    return ScriptType.CSharp;
-                case "js":
-                    return ScriptType.Javascript;
-                case "ts":
-                    return ScriptType.TypeScript;
+                    return "CSharp";
```

Comment:
So you're first parsing the built ins, then falling back to language service?

Reply:
Yes!